lib/collection/src/shards/replica_set.rs

```diff
@@ -1575,24 +1610,38 @@ impl ShardReplicaSet {
                             wait
                         };
 
+                    let operation = operation.clone();
+
                     let local_update = async move {
                         local
                             .get()
-                            .update(operation.clone(), local_wait)
+                            .update(operation, local_wait)
                             .await
                             .map_err(|err| {
                                 let peer_id = err.remote_peer_id().unwrap_or(this_peer_id);
 
                                 (peer_id, err)
                             })
                     };
+
                     update_futures.push(Either::Right(local_update));
                 }
             }
+
+            for remote in active_remote_shards {
+                let operation = operation.clone();
+                update_futures.push(Either::Left(async move {
+                    remote
+                        .update(operation, wait)
+                        .await
+                        .map_err(|err| (remote.peer_id, err))
+                }));
+            }
+
             match self.shared_storage_config.update_concurrency {
                 Some(concurrency) => {
                     futures::stream::iter(update_futures)
-                        .buffered(concurrency.get())
+                        .buffer_unordered(concurrency.get())
```

Comment:
Just an idea if we do not need the order, we could also improve the other branch using `join_all` later.

The doc says:

```
join_all will switch to the more powerful FuturesOrdered for performance reasons if the number of futures is large. You may want to look into using it or it's counterpart FuturesUnordered directly.
```

Reply:
Good point. I'll include this in the "fix potential flakiness of the test that I've "fixed" PR soon.